app/webpacker/components/Media/MediaTable.jsx

```diff
@@ -0,0 +1,254 @@
+import React, { useEffect, useState } from 'react';
+import _ from 'lodash';
+import { Table, Confirm, Modal, Button } from 'semantic-ui-react';
+import { competitionUrl, apiV0Urls } from '../../lib/requests/routes.js.erb';
+import { countries, continents, years } from '../../lib/wca-data.js.erb';
+import { dateRangeBetween } from '../../lib/helpers/media-table';
+import useLoadedData from '../../lib/hooks/useLoadedData';
+import useSaveAction from '../../lib/hooks/useSaveAction';
+import { DateTime } from 'luxon';
+import {
+  DropdownMenu,
+  DropdownItem,
+  DropdownHeader,
+  DropdownDivider,
+  Dropdown,
+} from 'semantic-ui-react'
+const countryOptions = _.map(countries.byIso2, (country) => ({
+  key: country.iso2,
+  text: country.name,
+  value: country.iso2,
+}));
+const continentOptions = continents.map((continent) => ({
+  key: continent.id,
+  text: continent.name,
+  value: continent.name,
+}));
+const yearOptions = [
+  {
+    key: 'all',
+    text: 'All Years',
+    value: 'All Years',
+  },
+  ...years.map((year) => ({
+    key: year,
+    text: year,
+    value: year,
+  })),
+];
+handleSort = (clickedColumn) => () => {
+  if (sortColumn !== clickedColumn) {
+    setSortColumn(clickedColumn);
+    setSortDirection('ascending');
+    setMediaCombined([...mediaCombined].sort((a, b) => a[clickedColumn] - b[clickedColumn]));
+  } else {
+    setMediaCombined([...mediaCombined].reverse());
+    setSortDirection(sortDirection === 'ascending' ? 'descending' : 'ascending');
+  }
+};
+
+export default function MediaTable({ isValidate }) {
+  const handleUpdateSuccess = () => {
+    setModalOpen(true);
+  };
+
+  const handleCloseModal = () => {
+    setModalOpen(false);
+  };
+  const [selectedYear, setSelectedYear] = useState(yearOptions[1].value);
+  const [selectedRegion, setSelectedRegion] = useState("All Regions");
+  const { save, saving } = useSaveAction();
+  const [mediaCombined, setMediaCombined] = useState([])
+  const type = isValidate ? 'pending' : 'accepted'
+  const { loading, error, data, sync } = useLoadedData(
+    apiV0Urls.media.list(type, selectedYear, selectedRegion)
+  );
+  const [modalOpen, setModalOpen] = useState(false);
+
+  useEffect(() => {
+    if (data) {
```

Comment:
nit: I'd use a `useMemo` to transform the data rather than a useEffect unless you also need to modify this data in `mediaCombined` after initial fetching.

Reply:
That is along the lines of what I suggested [here](https://github.com/thewca/worldcubeassociation.org/pull/8165#discussion_r1495037652); I didn't include a `useMemo` wrapper, but including one would be good.